src/python/pants/jvm/goals/coursier.py

```diff
@@ -97,24 +98,31 @@ async def coursier_generate_lockfile(
     jvm: JvmSubsystem,
     resolves_to_artifacts: JvmResolvesToArtifacts,
 ) -> CoursierGenerateLockfileResult:
+    requirements = ArtifactRequirements(resolves_to_artifacts.get(request.resolve, ()))
     resolved_lockfile = await Get(
         CoursierResolvedLockfile,
         # Note that it's legal to have a resolve with no artifacts.
-        ArtifactRequirements(resolves_to_artifacts.get(request.resolve, ())),
+        ArtifactRequirements,
+        requirements,
     )
-    resolved_lockfile_serialized = resolved_lockfile.to_serialized()
+    resolved_serialized_lockfile = resolved_lockfile.to_serialized()
+    metadata = JVMLockfileMetadata.new(requirements)
+    resolved_serialized_lockfile = metadata.add_header_to_lockfile(
+        resolved_serialized_lockfile, regenerate_command="./pants generate-lockfiles"
```

Comment:
TODO: check if this is correct after Eric's changes

Reply:
This _will_ be correct once the code gets moved, but presently it isn't.

Reply:
I'm close to putting up this change, so this is fine.